src/classes/TEST_RecurringDonationBuilder.cls

```diff
@@ -0,0 +1,363 @@
+/*
+    Copyright (c) 2019 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Recurring Donations Builder
+*
+*/
+@isTest
+public without sharing class TEST_RecurringDonationBuilder {
+    public static final Date DEFAULT_ESTABLISHED_DATE = Date.newInstance(1970,6,12);
+
+    private Map<String, Object> valuesByFieldName;
+
+    /***
+    * @description Constructor 
+    */
+    public TEST_RecurringDonationBuilder() {
+        valuesByFieldName = new Map<String, Object>();
+        
+        this.withName('RD' + UTIL_UnitTestData_TEST.getUniqueString());
+    }
+
+    /***
+    * @description Sets default values
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withDefaults() {
+        return this
+            .withAmount(2)
+            .withInstallments(2)
```

Comment:
`withInstallments` should not be in defaults because the definition of the field is that it only applies to `Fixed` length RD's. 

Overall this entire class has to be able to differentiate the data between RD1 and RD2 format recurring donations. There are a few ways to implement this. One might be new methods for `useRD1format` (the default) and `useRD2format`. Another is to check the RecurringDonationSettings.IsRd2Enabled value in the constructor and set a private var accordingly.  A third might be to use different methods such as `withDefaultsRD1()` and `withDefaultsRD2()`. My preference is the first option, but we'd need to make sure that all RD2 tests call `useRDformat()`.

In another comment I mentioned throwing an exception (could be `system.assert(false)` call too) if an RD1 only method (such as `withInstallments()` is called when in RD2 format mode. I think this would be very helpful in ensuring that RD test data is properly constructed for test methods.

Reply:
I have removed `withDefaults()` and moved specific RD field creation in specific test classes.